src/Essentials/src/Platform/Platform.ios.tvos.watchos.cs

```diff
@@ -138,5 +146,24 @@ internal static UIWindow GetCurrentWindow(bool throwIfNull = true)
 
 		internal static NSOperationQueue GetCurrentQueue() =>
 			NSOperationQueue.CurrentQueue ?? new NSOperationQueue();
+
+#if __IOS__
+		internal class UIPresentationControllerDelegate : UIAdaptivePresentationControllerDelegate
+		{
+			readonly Action dismissHandler;
+
+			internal UIPresentationControllerDelegate(Action dismissHandler)
+				=> this.dismissHandler = dismissHandler;
+
+			public override void DidDismiss(UIPresentationController presentationController) =>
+				dismissHandler?.Invoke();
```

Comment:
Can we set this to null ofter invoking? This way we don't get a double finish. Same with the Dispose invocation.

Reply:
yes, we can do it